src/core/WebAppManager.cpp

```diff
@@ -287,8 +288,7 @@ WebAppBase* WebAppManager::onLaunchUrl(const std::string& url, const std::string
       page->setEnableBackgroundRun(appDesc->isEnableBackgroundRun());
 
     app->setAppDescription(appDesc);
-    if (!app->isAglRoleType())
-	    app->setAglAppId(appDesc->id().c_str());
+    app->setAglAppId(appDesc->id().c_str());
```

Comment:
This should likely be moved eventually to a delegate, so WebAppManager does not know about the AGL specific bits.